src/hub_predtimechart/hub_config.py

```diff
@@ -78,6 +78,9 @@ def __init__(self, hub_dir: Path, ptc_config_file: Path):
         model_tasks_ele = tasks['rounds'][self.rounds_idx]['model_tasks'][self.model_tasks_idx]
         self.task_ids = sorted(model_tasks_ele['task_ids'].keys())
 
+        # set target_data_file_name
+        self.target_data_file_name = ptc_config.get('target_data_file_name')
```

Comment:
Very minor question: I noticed that here and in some other places above, we use `.get()`, while in other instances we use square brackets, e.g. for `self.target_date_col_name = ptc_config['target_date_col_name']`.  What's the difference between these constructions, and what's the reasoning behind different access methods in different places?

Reply:
get() returns None if the field is not present. this guarantees that the self.target_data_file_name variable will have a value for optional fields.